File: ExamenParcial-3S2/Pregunta3/UnitTest1/Requisito1.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ReSOSgame;
using System;

namespace UnitTest1
{
    [TestClass]
    public class Requisito1
    {
        private Tablero tablero;
        private Juego juego;
        [TestInitialize]
        public void Init()
        {
            tablero = new Tablero(3);
            juego = new JuegoSimple(tablero);
        }

        [TestMethod]
        [ExpectedException(typeof(Exception))]
        public void OutOfGridX()
        {
            juego.Jugar(5, 2,Tablero.Cell.S);
        }
        [TestMethod]
        [ExpectedException(typeof(Exception))]
        public void OutOfGridY()
        {
            juego.Jugar(3, 7,Tablero.Cell.S);
        }
        [TestMethod]
        public void AlreadyOccupied()
        {
            juego.Jugar(3, 3, Tablero.Cell.S);
            juego.Jugar(3, 3, Tablero.Cell.O);
            //se va a comprobar que el movimiento de O no se haya hecho y solo se haya realizado el movimiento X
            Assert.AreEqual(Tablero.Cell.S, tablero.GetCell(3, 3));
        }
    }
}

File: ExamenParcial-CC3S2/Pregunta2/src/test/java/Requisito1.java
import TicTacToe.TicTacToe;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;


public class Requisito1 {
    @Test
    public void OutOfGridXAxis(){
        assertThatExceptionOfType(RuntimeException.class).isThrownBy(
                ()->new TicTacToe().jugar(5,2));
    }
    @Test
    public void OutOfGridYAxis(){
        assertThatExceptionOfType(RuntimeException.class).isThrownBy(
                ()->new TicTacToe().jugar(1,7));
    }
    @Test
    public void AlreadyOccupied()
    {
        var ticTacToe = new TicTacToe();
        ticTacToe.jugar(3,3);
        ticTacToe.jugar(3,3);
        //se va a comprobar que el movimiento de O no se haya hecho y solo se haya realizado el movimiento X
        assertThat(ticTacToe.getCell(3,3)).isEqualTo(TicTacToe.Cell.CROSS);
    }
}
